frontend/src/metabase/core/components/AutocompleteInput/AutocompleteInput.unit.spec.tsx

```diff
@@ -57,4 +75,49 @@ describe("AutocompleteInput", () => {
 
     expect(onChange).toHaveBeenCalledWith("Orange");
   });
+
+  it("supports custom filtering functions", async () => {
+    const filterFn = (value: string | undefined, options: string[]) => {
+      if (value && options) {
+        return options.filter(
+          option =>
+            !option.toLocaleLowerCase().includes(value.toLocaleLowerCase()),
+        );
+      }
+      return [];
+    };
+
+    const { input, getOptions } = setup({
+      value: "or",
+      filterFn,
+    });
+
+    fireEvent.click(input);
```

Comment:
`fireEvent` is discouraged since it just dispatches a raw DOM event. `userEvent.click` is preferred